src/solvers/time_solver.C

```diff
@@ -124,9 +135,39 @@ void TimeSolver::integrate_adjoint_sensitivity(const QoISet & qois, const Parame
   return;
 }
 
-Real TimeSolver::last_complete_deltat()
+void TimeSolver::integrate_adjoint_refinement_error_estimate
+  (AdjointRefinementEstimator & adjoint_refinement_error_estimator,
+   ErrorVector & QoI_elementwise_error,
+   std::vector<Real *> /* QoI_time_instant */)
+{
+  // Make sure the system::qoi_error_estimates vector is of the same size as system::qoi
+  if(_system.qoi_error_estimates.size() != _system.qoi.size())
+      _system.qoi_error_estimates.resize(_system.qoi.size());
+
+  // Base class assumes a direct steady state error estimate
```

Comment:
So let's move this to SteadySolver, and put libmesh_not_implemented() in the base class?  Better to get an error message than an incorrect solution if someone's got a third-party derived class or if we miss an implementation in one of our own derived classes.

Reply:
We will need to do this for the other integrate_timestep functions as well, right ?

Reply:
Right.

Reply:
Done.